src/formatters/deployResultFormatter.ts

```diff
@@ -71,9 +71,11 @@ export class DeployResultFormatter extends ResultFormatter {
       const canceledByName = getString(this.result, 'response.canceledByName', 'unknown');
       throw new SfdxError(messages.getMessage('deployCanceled', [canceledByName]), 'DeployFailed');
     }
-    this.displaySuccesses();
-    this.displayFailures();
-    this.displayTestResults();
+    setTimeout(() => {
+      this.displaySuccesses();
+      this.displayFailures();
+      this.displayTestResults();
+    }, 500);
```

Comment:
you might need to make the `progress` method async and wrap everything inside of it in a Promise, and await that Promise